### PR TITLE
chore(ci): improve release production flow

### PR DESCRIPTION
**Pull request by yeager-eren**
# Summary

The goal of this task is making the publishing smoother and more easier then we can delegate the release to whole team.
I tried to simplify the process to hide some technical details and triggering the production flow using a single command. 

# Changes

- Adding a separate step after publishing the libraries to open a PR to merge `main` into `next`.
- Fix not running workflow twice on production.
- Previously, we should run two commands `release-prod` and `post-release-prod`. It only needs run the first command now.

# How did you test this change?

I added a new doc in `docs/release.md` which is an overview what is happening during a release. The first section has been moved from `Contributing.md` so you can only check the last section which is `How you can release a new version?`

For testing changes you will need to fork and setup its environments. You can find a guide on `docs/fork.md`.

Testing scenarios:

- Making a pull request on `next` then merge the PR, it should publishing `next` versions correctly. you making a change on `embedded` or any other libs that we have.
- After merging PR into `next`, You can run `yarn release-prod` it should trigger the production Github Actions and publish production.

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

**Comment by samobasquiat** [comment]
In `scripts/publish/utils.mjs`, we handle errors for conditions like `alreadyPublishedPackages`, `alreadyHasGithubReleasePackages`, and `alreadyHasGitTagPackages`. However, in `scripts/deploy/utils.mjs`, this error handling is missing. Is there a reason for this difference?

**Review by samobasquiat — CHANGES_REQUESTED** [review]

**Comment by yeager-eren** [comment]
> It's not related to this PR but In `scripts/publish/utils.mjs`, we handle errors for conditions like `alreadyPublishedPackages`, `alreadyHasGithubReleasePackages`, and `alreadyHasGitTagPackages` just for the main branch not for both main and next. Why?

Git tagging and making a Github release is only needed for production releases so they would be run on main only.
Checking packages on NPM is only a safety solution to make sure we don't run anything if certain situation hasn't been met since it can affect the users if we do anything wrong. For `next` version, that would simply fails the workflow and if anything bad happens, by fixing and increasing version it will be fixed usually.
So the tradeoff here is for production we need safety, for `next` we don't so by disabling this feature it can speed up the workflow's running.

File: scripts/common/utils.mjs [hunk 1]
```diff
@@ -92,13 +92,3 @@ export function getEnvWithFallback(name) {
...
 }
-
-/**
- * Opposite of `generateTagName`
- *
- * @param {string} pkgNameWithoutScope
- * @returns
- */
-export function tagNameToPkgName(pkgNameWithoutScope) {
-  return `${NPM_ORG_NAME}/${pkgNameWithoutScope}`;
```
[the remark this answers is not in the document]
Comment (yeager-eren) [reply scripts/common/utils.mjs:103]: Thanks. fixed.

**Review by samobasquiat — APPROVED** [review]
LGTM. the `.github/PUBLISH_TEMPLATE.md` file should be updated.

**Pull request merged**